src/Hakyll/Web/Template/Internal.hs

```diff
@@ -157,11 +158,14 @@ applyTemplate' tes name context x = go tes `catchError` handler
 
     applyElem Escaped = return "$"
 
-    applyElem (If e t mf) = (applyExpr e >> go t) `catchError` handler
+    applyElem (If e t mf) = do
+        c <- (applyExpr e >> return True) `compilerCatch` handler
+        if c
+            then go t
+            else maybe (return "") go mf
       where
-        handler _ = case mf of
-            Nothing -> return ""
-            Just f  -> go f
+        handler Error es = compilerThrow es
```

Comment:
I've had a quick google search for `field` and `fail` usage scenarios, and found none that were offending. Or actually, [this one](//github.com/jaspervdj/hakyll/pull/348#issuecomment-126206137).